scripts/30mini-iso-menu

```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+
+PREREQ=""
+
+prereqs()
+{
+       echo "$PREREQ"
+}
+
+case $1 in
+# get pre-requisites
+prereqs)
+       prereqs
+       exit 0
+       ;;
+esac
+
+. /scripts/casper-functions
+. /scripts/casper-helpers
+
+mountpoint=/cdrom
+
+iso_chooser_step1() {
+    # download JSON of simplestreams for showing the list of ISOs we might
+    # chain-boot to, hand that off to the menu, look what the choice was,
+    # use the requested ISO size to figure out if we have the memory or not,
+    # and kexec to reserve that memory
+
+    echo "###########################################"
+    echo "iso_chooser_step1"
+    echo "###########################################"
+
+    /usr/sbin/agetty --skip-login \
+        --login-program /usr/bin/iso-menu-session \
+        tty1 linux
+
+    if [ ! -f /out.vars ] ; then
+        echo "menu failure, debug shell"
+        /bin/sh
+    fi
+
+    . /out.vars
+
+    dev=$(blkid --match-token LABEL=ISOIMAGE | cut -d: -f1)
+
+    modprobe isofs
+    mount -o ro "$dev" "${mountpoint}"
+
+    configure_networking
+
+    echo "Loading $MEDIA_LABEL ..."
+
+    pmem_directive="memmap=$(/usr/bin/get_memmap_directive $MEDIA_SIZE)"
```

Comment:
This isn't run under set -e afaics, so a failure here won't halt the boot? It probably should?

Reply:
Done